File: app/src/test/java/org/schulcloud/mobile/data/model/FileTest.java
package org.schulcloud.mobile.data.model;

import org.junit.Before;
import org.junit.Test;
import static org.junit.Assert.*;

public class FileTest {

    private static final String KEY = "key";
    private static final String NAME = "TestFile.jpg";
    private static final String PATH = "TEST_PATH";
    private static final String SIZE = "200";
    private static final String TYPE = "IMAGE";
    private static final String THUMBNAIL = "TEST_THUMBNAIL";

    private File file;

    @Before
    public void setUp() {
        file = createNewFile();
    }

    @Test
    public void testGetProperties() {
        assertEquals(file.key, KEY);
        assertEquals(file.name, NAME);
        assertEquals(file.path, PATH);
        assertEquals(file.size, SIZE);
        assertEquals(file.type, TYPE);
        assertEquals(file.thumbnail, THUMBNAIL);
    }

    public static File createNewFile() {
        File file = new File();
        file.key = KEY;
        file.name = NAME;
        file.path = PATH;
        file.size = SIZE;
        file.type = TYPE;
        file.thumbnail = THUMBNAIL;

        return file;
    }

}

File: app/src/test/java/org/schulcloud/mobile/models/file/FileTest.kt
package org.schulcloud.mobile.models.file

import org.junit.Assert.*
import org.junit.Before
import org.junit.Test

class FileTest {
    private companion object {
        const val KEY = "key"
        const val NAME = "name"
        const val PATH = "path"
        const val SIZE = 100L
        const val TYPE = "type"
        const val THUMBNAIL = "thumbnail"
        const val FLATFILENAME = "flatName"
        const val ID = "key"
    }

    private val newFile: File
        get() = File().apply {
            key = KEY
            name = NAME
            path = PATH
            size = SIZE
            type = TYPE
            thumbnail = THUMBNAIL
            flatFileName = FLATFILENAME

        }
    private lateinit var file: File

    @Before
    fun setUp() {
        file = newFile
    }

    @Test
    fun testGetProperties() {
        assertEquals(KEY, file.key)
        assertEquals(NAME, file.name)
        assertEquals(PATH, file.path)
        assertEquals(SIZE, file.size)
        assertEquals(TYPE, file.type)
        assertEquals(THUMBNAIL, file.thumbnail)
        assertEquals(FLATFILENAME, file.flatFileName)
    }

    @Test
    fun testIdEqualsKey() {
        assertEquals(ID, file.id)
    }
}
